implementation/src/main/java/io/smallrye/config/ConfigSourceInterceptorFactory.java

```diff
@@ -0,0 +1,47 @@
+package io.smallrye.config;
+
+/**
+ * This ConfigSourceInterceptorFactory allows to initialize a {@link ConfigSourceInterceptor}, with access to the
+ * current {@link ConfigSourceInterceptorContext}.
+ *
+ * Interceptors in the chain are initialized in priority order and the current
+ * {@link ConfigSourceInterceptorContext} contains the current interceptor, plus all other interceptors already
+ * initialized.
+ */
+public interface ConfigSourceInterceptorFactory {
+    /**
+     * Gets the {@link ConfigSourceInterceptor} from the ConfigSourceInterceptorFactory. Implementations of this
+     * method must provide the instance of the {@link ConfigSourceInterceptor} to add into the Config Interceptor Chain.
+     *
+     * @param context the current {@link ConfigSourceInterceptorContext} with the interceptors already initialized.
+     * @return the {@link ConfigSourceInterceptor} to add to Config Interceptor Chain and initialize.
+     */
+    ConfigSourceInterceptor getInterceptor(final ConfigSourceInterceptorContext context);
+
+    /**
+     * Gets the Class of the Interceptor.
+     *
+     * This is required, because the interceptor priority needs to be sorted before doing initialization.
+     *
+     * @return the Class of the {@link ConfigSourceInterceptor}
+     */
+    Class<? extends ConfigSourceInterceptor> getInterceptorClass();
+
+    final class Delegate implements ConfigSourceInterceptorFactory {
```

Comment:
This should be static, I think.

Reply:
But interface members are also (almost) always public, and this is no exception.  It probably shouldn't be part of the API.

Reply:
Ok, let me move that to a separate class.